cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/MLSClientImpl.kt

```diff
@@ -137,22 +140,41 @@ class MLSClientImpl(
         return applicationMessage
     }
 
+    @Suppress("TooGenericExceptionCaught")
     override suspend fun decryptMessage(groupId: MLSGroupId, message: ApplicationMessage): List<DecryptedMessageBundle> {
-        val decryptedMessage = coreCrypto.decryptMessage(
-            groupId.decodeBase64Bytes(),
-            message
-        )
+        var decryptedMessage: DecryptedMessage? = null
+
+        coreCrypto.transaction(object : CoreCryptoCommand {
+            override suspend fun execute(context: CoreCryptoContext) {
+                try {
+                    val result = context.decryptMessage(
+                        groupId.decodeBase64Bytes(),
+                        message
+                    )
+                    decryptedMessage = result
+
+                } catch (throwable: Throwable) {
+                    val isBufferedFutureError = throwable.cause is MlsException.BufferedFutureMessage ||
+                            throwable.message?.contains("BufferedFutureMessage") == true ||
```

Comment:
Why do you need this? Isn't it already covered by `Throwable.cause is MlsException.BufferedFutureMessage`?

Reply:
Changed to proper error checking